### PR TITLE
CONTRIBUTING: how to build a package

### PR DESCRIPTION
**Pull request by mtojek**
This PR updates CONTRIBUTING docs with description on how to build/migrate an integration.

Preview: https://github.com/mtojek/package-registry/blob/how-to-build-package/CONTRIBUTING.md

I will mention few reviewers to verify if all steps are correct (agent, Kibana, etc.) 

Issue: https://github.com/elastic/package-registry/issues/220

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,29 +6,300 @@ This page is intended for contributors to the registry and packages.
...
+The dataset template consists of:
 
 * An alias templates (or the fields.yml to create it)
 * Zero or more ingest pipelines
 * An Elastic Agent config template
 
-## How to build a package
+### Migration from Beats
+
+A defined importing procedure used to transform both, Filebeat and Metricbeat modules, related to
```
Comment (exekias) [inline CONTRIBUTING.md:38]:
```suggestion
A defined importing procedure used to transform both Filebeat and Metricbeat modules, related to
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,29 +6,300 @@ This page is intended for contributors to the registry and packages.
...
+    1. Clone/refresh the following repositories:
+        * https://github.com/elastic/beats
+        * https://github.com/elastic/ecs
+        * https://github.com/elastic/eui
+        * https://github.com/elastic/kibana
+
+       Make sure you don't have any manual changes applied as they will reflect on the integration.
+    2. Clone/refresh the Elastic Package Registry (EPR) to always use the latest version of the script:
+        * https://github.com/elastic/package-registry
+    3. Make sure you've the `mage` tool installed.
```
Comment (exekias) [inline CONTRIBUTING.md:95]: nit: how about adding the command line for this?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -6,29 +6,300 @@ This page is intended for contributors to the registry and packages.
...
+3. Boot up required dependencies:
+    1. Elasticseach instance:
+        * Kibana's dependency
+    2. Kibana instance:
+        * used to migrate dashboards, if not available, you can skip the generation (`SKIP_KIBANA=true`)
+
+    _Hint_. There is dockerized environment in beats (`cd testing/environments`). Boot it up with the following command:
+    `docker-compose -f snapshot.yml -f local.yml up --force-recreate elasticsearch kibana`.
+4. Create a new branch for the integration in the EPR project (diverge from master).
```
Comment (exekias) [inline CONTRIBUTING.md:104]:
```suggestion
4. Create a new branch for the integration in package-registry repo (diverge from master).
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -6,29 +6,300 @@ This page is intended for contributors to the registry and packages.
...
+   $ cd $GOPATH/src/github.com/elastic/beats/x-pack/elastic-agent
+   $ PLATFORMS=darwin mage package
+    ```
+
+   Unpack the distribution you'd like to use (e.g. tar.gz):
+   ```bash
+   $ cd build/distributions/
+   $ tar xzf elastic-agent-8.0.0-darwin-x86_64.tar.gz
+   $ cd elastic-agent-8.0.0-darwin-x86_64/
+   ```
```
Comment (exekias) [inline CONTRIBUTING.md:286]: I think this is fine, wondering if we have a snapshot image being produced already? cc @ph @ruflin

File: CONTRIBUTING.md (hunk 2 above)
Comment (mtojek) [reply CONTRIBUTING.md]: Fixed. Added a one-liner for mage.

**Comment by mtojek** [comment]
> This is great, thank you!
> 
> Would it make sense to add a note to create an issue per integration explaining every manual change that was needed? That should help us improve the script later

Good idea, done.

**Comment by mtojek** [comment]
Thank you @exekias for the approval. I'm looking forward to see comments from other reviewers too.

File: CONTRIBUTING.md (hunk 4 above)
Comment (ph) [reply CONTRIBUTING.md:308]: I will take a look and get back to you @exekias

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -6,29 +6,322 @@ This page is intended for contributors to the registry and packages.
...
+
+A defined importing procedure used to transform both Filebeat and Metricbeat modules, related to
+the same observed product, into a single integration. The integration contains extracted dataset configuration of beat
+modules, hence no modules are required to exist anymore.
+
+## Package structure
+
+### Elements
+
+Link: https://github.com/elastic/package-registry/blob/master/ASSETS.md
```
Comment (ruflin) [inline CONTRIBUTING.md:46]: I will have to update that with the most recent details one day ;-)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -6,29 +6,322 @@ This page is intended for contributors to the registry and packages.
...
+The following packages can be considered as reference points for all integrations.
+
+#### Integration: reference
+
+Link: https://github.com/elastic/package-registry/tree/master/dev/packages/example/reference-1.0.0
+
+The directory contains mandatory manifest files defining the integration and its datasets. All manifests have fields
+annotated with comments to better understand their goals.
+
+_Keep in mind that this package doesn't contain all file resources (images, screenshots, icons) referenced in manifests.
```
Comment (ruflin) [inline CONTRIBUTING.md:59]: Ups, we should add those. I plan to have validation for these in the future.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -6,29 +6,322 @@ This page is intended for contributors to the registry and packages.
...
+4. Write README template file for the integration.
+
+    The README template is used to render the final README file including exported fields. The template should be placed
+    in the `dev/beats/import-beats-resources/docs/<integration-name>/docs/README.md`.
+
+    Review the MySQL docs template to see how to use template functions (e.g. `{{fields "dataset-name"}}`)
+
+5. Review fields file and exported fields in docs.
+
+    The fields files (`package-fields.yml`, `fields.yml` and `ecs.yml`) in the package were created from original
```
Comment (ruflin) [inline CONTRIBUTING.md:178]: Could you add an additional note here that the sum of all the fields.yml for a dataset should only contain the fields that are really used by the dataset? For example not just have all the ecs fields that exist?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -6,29 +6,322 @@ This page is intended for contributors to the registry and packages.
...
+4. Start Elasticsearch instance:
+
+    ```bash
+   $ yarn es snapshot -E xpack.security.authc.api_key.enabled=true 
+   ```
+
+5. Start Kibana with enabled Ingest Manager:
+
+    ```bash
+   $ yarn start --xpack.ingestManager.enabled=true --xpack.ingestManager.epm.enabled=true --xpack.ingestManager.fleet.enabled=true --xpack.ingestManager.epm.registryUrl=http://localhost:8080/
```
Comment (ruflin) [inline CONTRIBUTING.md:294]: Most users new to KB will forget to run `yarn kbn bootstrap` ;-)

File: CONTRIBUTING.md (hunk 7 above)
Comment (mtojek) [reply CONTRIBUTING.md:180]: Right, done.

File: CONTRIBUTING.md (hunk 8 above)
Comment (mtojek) [reply CONTRIBUTING.md:298]: Thanks, added.

**Pull request merged**